app_builder/lib/app_builder/macos.ex

```diff
@@ -0,0 +1,242 @@
+defmodule AppBuilder.MacOS do
+  @moduledoc false
+
+  import AppBuilder.Utils
+
+  def build_mac_app_dmg(release, options) do
+    {codesign, options} = Keyword.pop(options, :codesign)
+    {notarize, options} = Keyword.pop(options, :notarize)
+
+    release = build_mac_app(release, options)
+
+    app_name = Keyword.fetch!(options, :name)
+    File.rm_rf!("tmp/dmg")
+    File.mkdir_p!("tmp/dmg")
+    File.ln_s!("/Applications", "tmp/dmg/Applications")
+
+    File.cp_r!(
+      Path.join([Mix.Project.build_path(), "rel", "#{app_name}.app"]),
+      "tmp/dmg/#{app_name}.app"
+    )
+
+    to_sign =
+      "tmp/dmg/#{app_name}.app/**"
+      |> Path.wildcard()
+      |> Enum.filter(fn file ->
+        stat = File.lstat!(file)
+        Bitwise.band(0o100, stat.mode) != 0 and stat.type == :regular
+      end)
+
+    to_sign = to_sign ++ ["tmp/dmg/#{app_name}.app"]
+
+    if codesign do
+      entitlements_path = "tmp/entitlements.plist"
+
+      File.write!(entitlements_path, """
+      <?xml version="1.0" encoding="UTF-8"?>
+      <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+      <plist version="1.0">
+        <dict>
+        <key>com.apple.security.cs.allow-jit</key>
+        <true/>
+        <key>com.apple.security.cs.allow-unsigned-executable-memory</key>
+        <true/>
+        <key>com.apple.security.cs.allow-dyld-environment-variables</key>
```

Comment:
both wx & openssl are statically compiled so I believe we don't even need this entitlement for them. I though I'd leave it though in case someone Mix installs a dependency with native code that would take advantage of it.

Reply:
The static build is definitely very cool, the way to go. Just the build time and the nifs are troubling. Would be cool to have a repository of pre-built binaries for erlang on different platforms and versions. Having different versions available would also make building the mobile apps easier as they could fetch the correct runtime matching your local erlang version that is used to compile the beam files.

Reply:
Yes, a repository of builds for different targets is sorely needed. :) We need it, elixir-desktop needs it, [burrito](https://github.com/burrito-elixir/burrito) needs it too! (cc @doawoo @quinnwilton). I think it would be ideal to have as many official builds as possible. OTP team is keen with moving forward with https://github.com/erlang/otp/pull/5036 so we'd at least have Mac ones (just x86_64 until GitHub adds ARM workers though). I believe they would be keen to have builds for popular Linux distros too. Or maybe there's one Linux build with as many things statically compiled as possible so that it works on all distros. (Or, you know, actually two, one against gcc and one against musl)